apps/agent-runner/internal/server/liveness_test.go

```
@@ -0,0 +1,41 @@
+	// connection without explicitly closing its socket.
+	waitFor(t, 2*time.Second, func() bool {
+		runner.lifecycleMu.Lock()
+		defer runner.lifecycleMu.Unlock()
+		return len(runner.connections) == 0
+	})
+	if runner.manager.ActiveCount() != 1 {
+		t.Fatalf("stale socket cleanup canceled active execution")
+	}
+
+	reconnected := dialAndHandshakeExpectHealth(t, httpServer.URL, 1, "liveness")
+	defer reconnected.Close()
```

golangci-lint (2.13.2)

[error] Error return value of `reconnected.Close` is not checked

(errcheck)